src/passes/OptimizeInstructions.cpp

```diff
@@ -572,6 +571,14 @@ struct OptimizeInstructions
               (op = makeUnsignedBinaryOp(binary->op)) != InvalidBinary &&
               Bits::getMaxBits(binary->left, this) <= 31) {
             binary->op = op;
+          } else if (c < 0 && c > std::numeric_limits<int32_t>::min() &&
+                     binary->op == DivUInt32) {
```

Comment:
This seems logically disconnected from the previous code, so better not to connect them with an `else`.
```suggestion
          }
          if (c < 0 && c > std::numeric_limits<int32_t>::min() &&
              binary->op == DivUInt32) {
```